### DISCUSSION TITLE
Do we need bootstrap.js when exporting multiple modules?

### DISCUSSION THREAD

**Discussion by jessgoldq4** [module-federation/module-federation-examples#1684] [Q&A]
I'm looking at the [Advanced API: Startup Code](https://github.com/module-federation/module-federation-examples/blob/master/startup-code/README.md) example, and I have a few questions:

1. If I update App2 to include a second button component, <Button2>, and I expose it as a module, do I have to include it in the component rendered in bootstrap.js?

i.e,

![image](https://user-images.githubusercontent.com/90337173/156044883-169eac06-cdf5-43b8-8bb3-9240a84776c5.png)

![image](https://user-images.githubusercontent.com/90337173/156044840-f69604b7-ff78-433a-9ee8-45f5e5c43704.png)

![image](https://user-images.githubusercontent.com/90337173/156043541-f22fe678-fa1f-4838-930a-90fd39af16f4.png)

It seems like it's unnecessary; when I removed `<LocalButton2 />` from App2's App.js, I can still see <Button2> in the host app:

![image](https://user-images.githubusercontent.com/90337173/156043852-c8a0a58e-c712-44fe-9c73-1a999aa3d623.png)

which leads me to my next question...

---

2. If we're exporting individual components, and we don't need to include each component in App2's App.js, can we just return an import an empty <App> component into App2's bootstrap.js? It seems to work:

![image](https://user-images.githubusercontent.com/90337173/156045141-d8abf16c-5f2e-4a66-8b97-49b854c78499.png)

It also seems like leaving bootstrap.js in App2 completely empty works too: 

![image](https://user-images.githubusercontent.com/90337173/156044383-c46ca879-3860-49b9-9510-f3a91748b128.png)

What is the best practice here? Thanks

**Comment by ScriptedAlchemy** [comment]
No only needed on entry points of main apps themselves
